src/main/java/de/tum/in/www1/artemis/web/rest/programming/ProgrammingExerciseResource.java

```diff
@@ -878,4 +879,18 @@ public ResponseEntity<BuildLogStatisticsDTO> getBuildLogStatistics(@PathVariable
         return ResponseEntity.ok(buildLogStatistics);
     }
 
+    /**
+     * GET programming-exercises/checkout-directories
+     *
+     * @return a DTO containing the checkout directories for the exercise, solution, and test repository for the requested programming language.
+     */
+    @GetMapping("programming-exercises/repository-checkout-directories")
+    @EnforceAtLeastTutor
```

Comment:
Do tutors need this information? Is that displayed in the exercise view to tutors as well?

Reply:
I only had it enabled for editors at first and changed it to tutor since we do not really make a difference on the page, and it is no sensible data. But it probably makes the life for tutors easier if they only see what they should really need 👍